Create configuration file

michaella23 opened the pull request:
## Description

This PR adds a Configuration Markdown file to help users utilize this copilot-extension-template. Currently, there is text only. I wanted to know if you have a preference for file organization of screenshot images. Can I use the same images from the blog post? I have a few more questions, commented in the code on lines 12, 45, and 103.

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [x] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] 📦 Chore
- [ ] ⏩ Revert

## Related Tickets & Documents

Relates to #16

## Mobile & Desktop Screenshots/Recordings

N/A

## Steps to QA

N/A

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📜 CONTRIBUTING.md
- [ ] 🙅 no documentation needed
- [x] ⚙️ CONFIGURATION.md

## [optional] Are there any post-deployment tasks we need to perform?

N/A

## [optional] What gif best describes this PR or how it makes you feel?

![animated question mark with eyes rolling around](https://i.pinimg.com/originals/1c/13/fe/1c13fef08fca46438a9f364bdfa3458d.gif)

<!--
  For Work In Progress Pull Requests, please use the Draft PR feature,
  see https://github.blog/2019-02-14-introducing-draft-pull-requests/ for further details.

  For a timely review/response, please avoid force-pushing additional
  commits if your PR already received reviews or comments.

  Before submitting a Pull Request, please ensure you've done the following:
  - 📖 Read the Open Sauced Contributing Guide: https://github.com/open-sauced/.github/blob/main/CONTRIBUTING.md.
  - 📖 Read the Open Sauced Code of Conduct: https://github.com/open-sauced/.github/blob/main/CODE_OF_CONDUCT.md.
  - 👷‍♀️ Create small PRs. In most cases, this will be possible.
  - ✅ Provide tests for your changes.
  - 📝 Use descriptive commit messages.
  - 📗 Update any related documentation and include any relevant screenshots.
-->

nickytonline commented on CONFIGURATION.md:
```diff
@@ -0,0 +1,122 @@
+1. Enter a **GitHub App name**, e.g. my-copilot-extension
+2. Enter a URL for the homepage. This can be the same as the test URL for now.
+3. Set the **Callback URL** (currently required). This can be the same as the test URL for now. 
+<!-- ***** If these (Homepage URL and Callback URL) are indicated in the first part of the article, how best to reference it here? ***** --> 
```
You can use the screen shot from the blog post. In fact you can save all the images from the relevant sections.

michaella23 replied:
Can I make a directory for the screenshots? Something like `/docs/images`?

nickytonline replied:
Whatever you think makes sense. That sounds good to me. If we create a docs folder, consider putting the markdown file in there in the root as well.

nickytonline commented on CONFIGURATION.md:
```diff
@@ -0,0 +1,122 @@
+
+<!-- screenshot Copilot chat responding,  -->
+
+It won't respond to your specific question, as that functionality has not been implemented. This is where you can explore the preview SDK or integrate with a third-party service to provide more meaningful responses.
+
+<!-- ***** Is this to be included as well? *****
```
So the only sections to omit from the blog post are the `Creating the Endpoint for Your Copilot Extension`, `A Real-World Example`, `Grab the GitHub Copilot Extension Template`, `Take a deep dive into Copilot extensions and the preview SDK` and the `Wrapping Up` sections of the blog post.

nickytonline reviewed (commented):
Let me know if anything else isn't clear @michaella23. Once the images are in, feel free to take it out of draft mode and we can iterate from there. Thanks for working on this!

nickytonline commented:
Also, currently in the README there is a getting started section.

![CleanShot 2024-11-03 at 08 18 56](https://github.com/user-attachments/assets/c8b8f15e-fcaf-4570-b79e-5305eb124d02)

We could link to your file in the README with something like, To get up and running with your development environment, see the [Development Guide](./), or something like that.

```diff
- ## Getting Started
+ ## Installation
+
+ Run the following commands to install and start the application locally

npm install
npm run dev
open http://localhost:3000

- Use tunneling to expose your localhost to the internet via [VS Code's Port forwarding](https://code.visualstudio.com/docs/editor/port-forwarding), [ngrok](https://ngrok.com), or similar.

- You'll need to create a GitHub app for the Copilot extension to work and use the URL from the from the tunneling above in your GitHub app configuration.

- For a full breakdown of how to get started, check out Nick's Blog post, [Creating a GitHub Copilot Extension: A Step-by-Step Guide](https://nickyt.co/copilot-extension-guide).
+
+ ## Development Environment
+
+ To get up and running with your development environment, see the [Development Guide](./)
```

michaella23 commented:
Hey @nickytonline! I've finished the next draft of the Development Guide. 

- [x] Created `CONFIGURATION.md`
- [x] Adapted text from article for this documentation
- [x] Added all relevant screenshots in a new directory - `./docs/images`
- [x] Moved `CONFIGURATION.md` to the root of the `docs` directory
- [x] Edited `README.md` to route the user to the Development Guide

Please let me know if there are any changes you'd like me to make!

nickytonline commented on docs/CONFIGURATION.md:
````diff
@@ -0,0 +1,125 @@
+
+To test your Copilot extension, you need to make it publicly accessible:
+
+1. If using Visual Studio Code (VS Code), enable [port forwarding](https://code.visualstudio.com/docs/editor/port-forwarding). Note that the port is private by default - a good thing - but for this use case you need to set it to public.
+
+![Port forwarding tab in the bottom dock of VS Code](./images/port_forwarding.png)
+
+Alternatively, use tools like [cloudflared](https://developers.cloudflare.com/cloudflare-one/connections/connect-networks/get-started/create-local-tunnel/) or [ngrok](https://ngrok.com/) to expose a public URL.
+
+In the provided code, the server is set up to run on port 3000:
+
+```
+const port = 3000;
+console.log(`Server is running on port ${port}`);
+
+serve({
+  fetch: app.fetch,
+  port,
+});
+```
````
This is how I wrote the blog post, but I'm going to update it to this.

```suggestion
1. If using Visual Studio Code (VS Code), enable [port forwarding](https://code.visualstudio.com/docs/editor/port-forwarding). Note that the port is private by default - a good thing - but for this use case you need to set it to public.

![Port forwarding tab in the bottom dock of VS Code](./images/port_forwarding.png)

Alternatively, use tools like [cloudflared](https://developers.cloudflare.com/cloudflare-one/connections/connect-networks/get-started/create-local-tunnel/) or [ngrok](https://ngrok.com/) to expose a public URL.

The application is set up to run on port 3000, so we'll forward port 3000.
```

nickytonline reviewed (changes requested):
One small change request to the docs, and also, let's renamed CONFIGURATION.md to SETUP.md or DEVELOPMENT_SETUP.md. I'm good with either.

Thanks again for working on this!

michaella23 commented:
> One small change request to the docs, and also, let's renamed CONFIGURATION.md to SETUP.md or DEVELOPMENT_SETUP.md.

Hey @nickytonline, I've made suggested changes. 🙂

nickytonline commented on README.md:
```diff
@@ -2,16 +2,16 @@
+## Development Environment
 
-For a full breakdown of how to get started, check out Nick's Blog post, [Creating a GitHub Copilot Extension: A Step-by-Step Guide](https://nickyt.co/copilot-extension-guide).
+To get up and running with your development environment, see the [Development Guide](./docs/CONFIGURATION.md).
```
```suggestion
To get up and running with your development environment, see the [Development Guide](./docs/DEVELOPMENT_SETUP.md).
```

nickytonline reviewed (approved):
Thanks for this @michaella23! :shipit:
